js/tooltips.js

```diff
@@ -0,0 +1,87 @@
+import Adapt from 'core/js/adapt';
+import _ from 'underscore';
+import logging from './logging';
+import TooltipView from './views/TooltipView';
+import TooltipModel from './models/TooltipModel';
+import Backbone from 'backbone';
+
+class TooltipController extends Backbone.Controller {
+
+  initialize() {
+    _.bindAll(this, 'onMouseOver');
+    this._tooltipData = {};
+    this._currentId = null;
+    this.listenTo(Adapt, 'adapt:preInitialize', this.onAdaptPreInitialize);
+  }
+
+  onAdaptPreInitialize() {
+    const config = this.getConfig();
+    if (config?._isEnabled === false) return;
+    this.attachToBody();
+    $(document).on('mouseover', '*', this.onMouseOver);
+  }
+
+  getConfig() {
+    return Adapt.course.get('_tooltips');
+  }
+
+  attachToBody() {
+    this.tooltipsView = this.tooltipsView || new TooltipView();
+    const $el = this.tooltipsView.$el;
+    if ($el[0].parentNode && $el.is(':last-child')) return;
```

Comment:
could `$el[0].parentNode` ever return undefined? Could guard it using the optional chaining operator ? `$el[0]?.parentNode` to make sure we dont get a type error of Cannot read properties of undefined